pyhealth/datasets/base_image_dataset.py

```diff
@@ -0,0 +1,141 @@
+import logging
+import os
+from abc import ABC
+from typing import Optional, Callable
+
+import pandas as pd
+from tqdm import tqdm
+
+from pyhealth.datasets.sample_dataset import SampleImageDataset
+
+logger = logging.getLogger(__name__)
+
+INFO_MSG = """
+dataset.patients:
+    - key: patient id
+    - value: a dict of features, labels, and other information
+"""
+
+
+class BaseImageDataset(ABC):
+    """Abstract base Image dataset class.
+
+    This abstract class defines a uniform interface for all image datasets.
+
+    Each specific dataset will be a subclass of this abstract class, which can then
+    be converted to samples dataset for different tasks by calling `self.set_task()`.
+
+    Args:
+        root: root directory of the raw data (should contain many csv files).
+        dataset_name: name of the dataset. Default is the name of the class.
+        dev: whether to enable dev mode (only use a small subset of the data).
+            Default is False.
+        refresh_cache: whether to refresh the cache; if true, the dataset will
+            be processed from scratch and the cache will be updated. Default is False.
+    """
+
+    def __init__(
+        self,
+        root: str,
+        dataset_name: Optional[str] = None,
+        dev: bool = False,
+        refresh_cache: bool = False,
+    ):
+        # base attributes
+        self.dataset_name = (
+            self.__class__.__name__ if dataset_name is None else dataset_name
+        )
+        self.root = root
+        # TODO: dev seems unnecessary for image and signal?
+        self.dev = dev
+        if dev:
+            logger.warning("WARNING: dev has no effect for image datasets.")
+        # TODO: refresh_cache seems unnecessary for image and signal?
+        self.refresh_cache = refresh_cache
+        if refresh_cache:
+            logger.warning("WARNING: refresh_cache has no effect for image datasets.")
+
+        self.metadata = pd.read_json(os.path.join(root, "metadata.jsonl"), lines=True)
```

Comment:
Is it common to have .jsonl for every image dataset? Maybe move it to covid19_xray.py